compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2610,23 +2580,19 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
   protected def addAccessorDefs(cls: Symbol, body: List[Tree])(using Context): List[Tree] =
     PrepareInlineable.addAccessorDefs(cls, body)
 
-  /** If this is a real class, make sure its first parent is a
-   *  constructor call. Cannot simply use a type. Overridden in ReTyper.
+  /** Turn a parent type into a constructor call where needed. This is the case where
+   *   - we are in a Scala class or module (not a Java class, nor a trait), and
+   *     - the parent symbol is a non-trait class, or
+   *     - the parent symbol is a trait that takes at least one (explicit or implicit) parameter
+   *       and the parent symbol is directly extended by the current class (i.e. not
+   *       extended by the superclass).
```

Comment:
> (i.e. not extended by the superclass)

I don't think it's equivalent in this case. This condition will also exclude examples like:
```scala
class C[X]
given C[Int] = new C[Int]

trait T[X](implicit ev: C[X])
class Base extends T
class Z extends Base with T
```
Since both `Z` and `Base` extend `T`.

Is this desired?

Reply:
Yes. `Z` will not invoke the constructor of `T`, so the reference should not be converted.

Reply:
Right, but then `result.symbol.info.takesImplicitParams` becomes false, since `result` has no symbol and the snippet fails with
```scala
-- [E056] Syntax Error: tests/pos/i15976.scala:19:26 ---------------------------
19 |class Z extends Base with T
   |                          ^
   |                          Missing type parameter for T
```
So I'm wondering if there might be a way around that.

Reply:
There's nothing that can be done here according to our design for trait inheritance.